pkg/email/parser.go

```diff
@@ -321,6 +347,8 @@ func strToCmd(str string) Command {
 		return CmdSet
 	case "regenerate":
 		return CmdRegenerate
+	case "skip", "skip:":
```

Comment:
We also need an opposite command. Un-undoable actions are not good.

We need some consistent way to change "bool" tags.

Also re consistency with subsystems, you used the set command to set subsystems. I though the idea is to use it to set all tags. Shouldn't we use "#syz set skip", "#syz set skip: false"?
Or otherwise should we remove "set" from subsystems command? It's strange to use an additional verbosity "set" only in 1 command, but not in other similar commands.

Reply:
Hmm, it's tricky.

- Generally we want to classify tags in groups (`subsystems`, later also `bug origin`) to avoid name collisions. It also makes it easier to update tags automatically -- just clear the whole group and set new labels. All unrelated tags automatically stay the same.
- For subsystems (and later maybe not just for them), we want to let users set multiple subsystem tags at the same time.
- At the same time, the `skip` (or, maybe better, `no-reminders`) tag doesn't seem to be worth a separate group. It's just a plain tag.
- For bug lists, we need to specify the title of the exact bug that was meant.

If we want to stick to `#syz set`, we might try to follow this approach.

**For plain tags** accept commands (a) for bug reports:
`#syz set no-reminders`
`#syz unset no-reminders` 

And (b) for bug lists:
`#syz set no-reminders for INFO: task hung in netdev_run_todo (2)`
`#syz unset no-reminders for INFO: task hung in netdev_run_todo (2)`

Technically we could let users set whatever plain tags they want (and store them just as a list, i.e. no explicit fields in `bug.Tags`).

**Grouped tags** (`subsystems`) are to be handled separately (e.g. we need to verify that the specified subsystem names exist). So we store them explicitly as `bug.Tags` fields.

For bug reports:
`#syz set subsystems: net, usb`

For `#syz unset` there doesn't seem to be any particular reason (it's equal to `#syz set subsystems:`).

For bug lists: we could have tried `#syz set subsystems for INFO: task hung in netdev_run_todo (2): net`, but I'm not sure it's needed that much now and also I'm not sure how well we could handle newlines due to 80 char line width limit in this case.

Wdyt?

Reply:
Maybe we can get some inspiration from the debian bug tracking system, it seems they have some consistent interface:

https://www.debian.org/doc/manuals/debian-bugs/ch-mailcontrol.en.html
https://www.debian.org/Bugs/server-control
https://www.debian.org/Bugs/server-control#tag
https://www.debian.org/Bugs/Developer#tags

The commands with titles and multiple colons look clumsy and hard to get right. Maybe we need to use bug numbers/hashes?

Reply:
In those links they seem to just include the bug key into the command string:
`tags bugnumber [ + | - | = ] tag [ tag ... ] [ + | - | = tag ... ] ]`

Given that we already have some interface, I think `+` and `-` are better be replaced by `#syz set` and `#syz unset`.
To use bugnummer is a good idea. I don't know if our long bug IDs are fine for that purpose, but we could at least use per-report bug references.

For plain tags (adds or removes a single one):
`#syz (un)set (<ref>)? tag`

For group tags (overrides them):
`#syz (un)set (<ref>)? group-name: sub-tag [sub-tag ...]`

Something like `#syz set <1> subsystems: net` or `#syz set <1> no-reminders`

```
Ref   Crashes Repro Title
<1>   785     Yes   possible deadlock in ovl_maybe_copy_up
                    https://syzkaller.appspot.com/bug?extid=c18f2f6a7b08c51e3025
<2>   442     Yes   possible deadlock in mnt_want_write (2)
                    https://syzkaller.appspot.com/bug?extid=b42fe626038981fb7bfa
<3>   24      Yes   BUG: unable to handle kernel paging request in take_dentry_name_snapshot
                    https://syzkaller.appspot.com/bug?extid=90392eaed540afcc8fc3
<4>   11      Yes   WARNING: locking bug in take_dentry_name_snapshot
                    https://syzkaller.appspot.com/bug?extid=5a195884ee3ad761db4e
```